File: data/readable_format/fortran/file2376.f90
! Define the module performance
module performance

  implicit none
  save

  private
  public :: perf_header, perf_footer, perf_numbers

contains

  subroutine perf_header()
    implicit none
    write(*,*)
    write(*,'("        # Photons    CPU time (sec)    Photons/sec  ")')
    write(*,'("      ----------------------------------------------")')
  end subroutine perf_header

  subroutine perf_footer()
    implicit none
    write(*,'("      ----------------------------------------------")')
    write(*,*)
  end subroutine perf_footer

  subroutine perf_numbers(count, time)
    implicit none
    integer, intent(in) :: count
    real, intent(in) :: time
    if(time < tiny(1.0)) then
      write(*,'(1X,3X,I12,3X,4X,F10.1,4X,4X,"   ...   ")') count, time
    else
      write(*,'(1X,3X,I12,3X,4X,F10.1,4X,4X,F9.2,4X)') count, time, real(count)/time
    end if
  end subroutine perf_numbers

end module performance

! Main program starts here
program test_performance
  use performance
  implicit none

  integer :: count
  real :: time

  ! Call the performance reporting subroutines
  call perf_header()
  count = 1000000
  time = 123.4
  call perf_numbers(count, time)
  count = 500000
  time = 0.0
  call perf_numbers(count, time)
  call perf_footer()

end program test_performance

File: data/readable_format/cpp/file2376.cpp
#include <iostream>
#include <iomanip>
#include <limits>
#include <cmath>
#include <sstream>
#include <string>

// Performance functions
namespace performance {

    void perf_header(std::ostream& os) {
        os << std::endl;
        os << "        # Photons    CPU time (sec)    Photons/sec  " << std::endl;
        os << "      ----------------------------------------------" << std::endl;
    }

    void perf_footer(std::ostream& os) {
        os << "      ----------------------------------------------" << std::endl;
        os << std::endl;
    }

    void perf_numbers(std::ostream& os, int count, double time) {
        if (time < std::numeric_limits<double>::min()) {
            os << std::setw(1) << " "
               << std::setw(3) << " "
               << std::setw(12) << count
               << std::setw(3) << " "
               << std::setw(4) << " "
               << std::setw(10) << std::fixed << std::setprecision(1) << time
               << std::setw(4) << " "
               << "   ...   " << std::endl;
        } else {
            os << std::setw(1) << " "
               << std::setw(3) << " "
               << std::setw(12) << count
               << std::setw(3) << " "
               << std::setw(4) << " "
               << std::setw(10) << std::fixed << std::setprecision(1) << time
               << std::setw(4) << " "
               << std::setw(9) << std::fixed << std::setprecision(2) << static_cast<double>(count) / time
               << std::setw(4) << " " << std::endl;
        }
    }
}

// Manual Test Function
void test_performance() {
    std::ostringstream test_stream;
    performance::perf_header(test_stream);
    performance::perf_numbers(test_stream, 1000000, 123.4);
    performance::perf_numbers(test_stream, 500000, 0.0);
    performance::perf_footer(test_stream);

    // Print the test output for manual verification
    std::cout << test_stream.str() << std::endl;
}

int main() {
    // Run the test
    test_performance();

    // Add your application logic or further tests here

    return 0;
}